CommonTools/PileupAlgos/src/PuppiContainer.cc

```diff
@@ -349,3 +350,22 @@ std::vector<double> const &PuppiContainer::puppiWeights() {
   }
   return fWeights;
 }
+// ------------------------------------------------------------------------------------------
+void PuppiContainer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
```

Comment:
Sorry I think I am confused. 
As far as I know, the fillDescriptions is the method that will create this *_cfi files correct? and also, you asked me to move all the descriptions definition from the PuppiProducer to PuppiAlgos/PuppiContainers. Then, if I only include the ParameterSet descriptions here (fillDescriptionsPuppiContainer), what is the benefit of having it here if it will not do anything?

Reply:
The parameters for PuppiAlgo (e.g.) are used inside PuppiAlgo.
If you define them inside PuppiProducer, then any time a parameter is added/removed/modified in PuppiAlgo then one has to go to PuppiProducer and modify the FillDescription method also there.
If instead inside ```PuppiProducer::fillDescriptions``` you call ```PuppiContainer::fillDescriptionsPuppiContainer(desc)```, which in its turn calls ```PuppiAlgo::fillDescriptionsPuppiAlgo(desc);``` (keep desc as argument), then whatever happens insidePuppiContainer and/or PuppiAlgo remains transparent to PuppiProducer